platform/src/main/java/org/stellar/anchor/platform/utils/TransactionHelper.java

```diff
@@ -66,10 +66,18 @@ static GetTransactionResponse toGetTransactionResponse(
       refunds = toRefunds(txn.getRefunds(), txn.getAmountInAsset());
     }
 
+    String sender = DEPOSIT.kind.equals(txn.getKind()) ? txn.getToAccount() : txn.getFromAccount();
+
     String amountInAsset = makeAsset(txn.getAmountInAsset(), assetService, txn);
     String amountOutAsset = makeAsset(txn.getAmountOutAsset(), assetService, txn);
     String amountFeeAsset = makeAsset(txn.getAmountFeeAsset(), assetService, txn);
     String amountExpectedAsset = makeAsset(null, assetService, txn);
+    Customers.CustomersBuilder builder =
+        Customers.builder().sender(StellarId.builder().account(sender).build());
+
+    if (WITHDRAWAL.kind.equals(txn.getKind()) && txn.getWithdrawAnchorAccount() != null) {
+      builder.receiver(StellarId.builder().account(txn.getWithdrawAnchorAccount()).build());
+    }
```

Comment:
You're assigning the businesses's destination account for receiving payments from users as `customer.receiver`, which is incorrect. The Stellar Id for the receiver should be the account authenticated via SEP-10, and usually this is also the source of the payment to the anchor.